Add Zeus as part of the main guide (new version)

VajraOfIndra opened the pull request:
(this PR supercedes #839 which had conflicts preventing to be merged)

#### What

This PR moves the Zeus bonus guide to the main part of the guide under the 'Lighnting' section.
It also update the guide to the latest versions of lndconnect and Zeus and to RaspiBolt v3 standards.

### Why

The Zeus mobile app connects directly to the node and allows to make on-chain and LN payments and operate the LN node while on-the-move. This use case is important enough to be part of the main guide rather than the bonus guide. 

Zeus is actively maintained, constantly adds new features as the LN space develops. It is also cross-platform and can be used by both Android and iOS users.

#### How

- Moved bonus guide to 'Lightning' section of the main guide.
- Changed title to "Mobile Wallet" to use a generic title rathern than a program name as is done in  the Bitcoin chapter.
- Updated guide: lndconnect v0.2, illustrations etc
- Updated the 'Index' and 'Lightning' pages to make reference to Zeus
- Updated the 'RTL' page to points to 'Zeus' as the next page of the guide
- Updated the '+ Lightning' page to remove the refreence to the Zeus bonus guide

#### Scope

- [x] significant change to core configuration
- [ ] independent bonus guide
- [ ] simple bug fix

#### Test & maintenance

Test: Follow the guide and test that Zeus is properly connecting to the node.

Maintenance: Low. Keep an eye on updates for lndconnect (although not maintained anymore)

#### Animated GIF (optional)

![zeus](https://media.giphy.com/media/Y4JPYsrJEk454pNbts/giphy-downsized.gif)

VajraOfIndra commented:
There were some conflicts on the existing PR so I created a clean new branch and closed the old one.
 
The only new thing is that I changed the title of the guide from "Zeus" to "Mobile Wallet" to keep it generic as is done in the Bitcoin chapter.

Note: It seems Zeus v0.6 will be released on Feb the 25th (source: https://github.com/ZeusLN/zeus/issues/836).

Stadicus commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+{: .no_toc }
+
+We install [Zeus](https://zeusln.app/){:target="_blank"}, a cross-platform mobile app that connect to your LN node over Tor.  
+Make payments with lightning or on-chain and manage your channels wihle you're on the go.
```
> manage your channels wihle you're on the go

"while"

VajraOfIndra replied:
Fxied!

Stadicus commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+### Create a lndconnect QR code
+
+lnconnect can generate a URI and create a QR code that can then be read by Zeus.
```
I usually prefer simple and  "active" instead of "passive" wording, e.g. 

> lnconnect generates a URI and displays it as a QR code that Zeus can read.

VajraOfIndra replied:
I'll try to keep that in mind, thanks

Stadicus commented on lightning-mobile-wallet.md:
````diff
@@ -0,0 +1,161 @@
+  $ lndconnect --host=abcdefg..............xyz.onion --port=8080
+  ```
+  
+* It will be a big QR code so maximize your terminal window and use CTRL - to shrink the code further to fit the screen
````
Missing comma: 

> It will be a big QR code, so maximize

VajraOfIndra replied:
Fixed

Stadicus commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+* Open Zeus and click on 'Get started'
+
+* The screen proposes several settings, we'll come back to them later, for now select 'Connect a node' at the top, and then '+ Add a new node'
```
Long sentence. Maybe:

> The screen proposes several settings; we'll return to them later. For now, select ...

VajraOfIndra replied:
Thinking about it... we never come back to the other options (recovering a wallet or creating a new wallet), so we probably can just delete "The screen proposes several settings, we'll come back to them later, for now" and start the sentence at "Select "Connect a nde"..." sounds good?

Stadicus replied:
Yep, makes sense

Stadicus commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+* Click on 'Use Tor'
+
+* Click on 'Scan LNDConnect config' and when prompted, allow Zeus to take pictures
```
Either +1 comma:

> Click on 'Scan LNDConnect config' and, when prompted, allow Zeus to take pictures

Or moving some words

> Click on 'Scan LNDConnect config' and allow Zeus to take pictures when prompted

Stadicus commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+* Scan the QR code generated earlier
+
+* Click on 'Save settings', Zeus is now connecting to your node, it might take a while the first time.
```
I'd split the sentences:

> Click on 'Save settings'. Zeus is now connecting to your node, **and** it might take a while the first time.

VajraOfIndra replied:
Good idea, fixed

Stadicus reviewed (commented):
Awesome guide, thanks a lot! 

I left some minor comments and nits, nothing ground-breaking, I think it's almost ready to merge.

VajraOfIndra commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+  $ tar -xvf lndconnect-linux-arm64-v0.2.0.tar.gz
+  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lndconnect-linux-arm64-v0.2.0/lndconnect
+  $ rm lndconnect-linux-arm64-v0.2.0.tar.gz
+  $ rm -R lndconnect-linux-arm64-v0.2.0
```
Good point. Moreover it's always risky to use `rm`.. better get rid of it.

VajraOfIndra commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+* Scan the QR code generated earlier
+
+* Click on 'Save settings', Zeus is now connecting to your node, it might take a while the first time.
+
```
I haven't tried v0.6 yet (I'm not ready to use pre-releases with my real node).. I assume the message will still be here. But we could test it when it comes out in a few days.

What warning message would you suggest?

Stadicus replied:
Something like

> Zeus will show a warning about "certificate verification". Because we use a Tor v3 hidden service to connect to LND, the connection is end-to-end encrypted by Tor and no additional certificate is needed. Click on "I understand, save my node config".

Stadicus replied:
There's no more warning when Tor is used. Makes sense and makes this issue moot.

VajraOfIndra commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+### Add a password
+
+For added security, you can add a password.
```
I added more context and a strong recommendation to set up the app password. I also changed the heading to "Secuirty".

Regarding where to find the option and the crashes, I propose we wait until v0.6 is out, like this we can tailor the wording of the guide to match the layout of the v0.6 app. Wdyt?

Stadicus replied:
Yes, that makes a lot of sense.

Stadicus replied:
No more warning in v0.6

VajraOfIndra commented on lightning-mobile-wallet.md:
```diff
@@ -0,0 +1,161 @@
+
+## Zeus in action
+
+Below is a list of Zeus existing (ticked) and coming soon (unticked) features:
```
Good point :) I also removed the parentheses entirely (it's an addiction, I'm working on it :)

VajraOfIndra commented on lightning.md:
```diff
@@ -25,4 +25,5 @@ Check out [Understanding the Lightning Network](https://bitcoinmagazine.com/tech
 To enable the Lightning Network on your RaspiBolt, we install LND, the "Lightning Network Daemon". 
 We then set up an automatic Static Channel Backup to protect ourselves in case of failure of the SSD drive. 
 We'll then add "Ride The Lightning", a web-based node management tool. 
+Finally, we'll install the Zeus mobile app, to make on-chain and LN payments and manage our node while we're on the go.
```
Fixed

Stadicus commented:
> What do you think about the title of the chapter that will be part of the left Menu...

Regarding "Generic" vs. "Application-specific" titles: I strongly prefer the generic titles, at least when there's a choice. This also affects the filenames, and we can replace individual applications in the core guide easier if necessary.

The one place where I prefer the Application title is "Bitcoin Core". There's not really a choice here, IMO, and it helps to differentiate from the generic "Bitcoin" section. OTOH, a title like "Bitcoin client" would also work.

Stadicus commented:
I resolved most comments, a few remain open for later finalization. I agree that we should wait for Zeus v0.6 to be released, finalize the guide, and then merge it.

Stadicus commented:
Yay, Zeus v0.6 has officially been released. I'm currently testing the guide and can provide screenshots. Some wording changes might be necessary, then let's publish!

VajraOfIndra commented:
> Yay, Zeus v0.6 has officially been released.

:grinning: :zap: 

> I'm currently testing the guide and can provide screenshots. Some wording changes might be necessary, then let's publish!

Yes, was working on updating the "Connect Zeus to your node" section to match the new process, but my phone refuses to scan the QR code..! :cry:   So I can't confirm steps beyond the QR code scanning step..

So far I reworded like this:

> 
> ### Connect Zeus to your node
> 
> * Open Zeus and select `Get started`
> 
> * Select `Connect to an lnd node`
> 
> * Tick the "Use Tor" box
> 
> * Select `Scan lndconnect config` and, if prompted, allow Zeus to take pictures
> 
> * Scan the QR code generated earlier
> 
> * Click on 'Save settings'. Zeus is now connecting to your node, and it might take a while the first time.

Regarding the certificate warning we wanted to add: 

> Zeus will show a warning about "certificate verification". Because we use a Tor v3 hidden service to connect to LND, the connection is end-to-end encrypted by Tor and no additional certificate is needed. Click on "I understand, save my node config"

... at what step does this need to be inserted, is it just after scanning the QR code?

I'll use your screenshots to update the section, or else please suggest changes.

Stadicus commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+
+* Click on 'Use Tor'
+
+* Click on 'Scan LNDConnect config' and allow Zeus to take pictures when prompted
```
s/take pictures/use the camera/

VajraOfIndra replied:
Fixed

Stadicus commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+
+* Scan the QR code generated earlier
+
+* Click on 'Save settings'. Zeus is now connecting to your node, and it might take a while the first time.
```
s/Save settings/SAVE NODE CONFIG/

VajraOfIndra replied:
Fixed

Stadicus commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+
+Anyone using Zeus on your phone has control over your node and all its funds. It is strongly recommended to set up a password for the app.
+
+* In the app, go to the settings and select 'Security' and set a passphrase (save your passphrase somewhere safe, e.g., your password manager).
```
Updated wording

> * In the app, tap on the Zeus icon in the top-left corner
> * Click on "Security" and "Set/Change Password" to enter a password or PIN
> * Save your password or PIN somewhere safe, e.g., in your password manager

VajraOfIndra replied:
Fixed

Stadicus reviewed (approved):
Just a few little wording changes, the overall guide is good to go! :rocket:

VajraOfIndra commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+On Android, download the Zeus app on F-Droid or Google Play.  
+
+On iOS, download it from the Apple App Store.
+
```
That's a great idea. I believe I spotted a scam app that looks like Zeus in the Google Play....

 I just feel a bit sad  that the website does not list the F-Droid app store which we should encourage our users to use above Google Play.

Would you be ok if I add a few words and link to the Zeus app in the F-Droid store? for example:

> Download the Zeus app for your mobile phone. Check the [Zeus website](https://zeusln.app/){:target="_blank"} for the direct download links to the Apple App Store or Google Play. Or get it from the [F-Droid store](https://f-droid.org/en/packages/app.zeusln.zeus/){:target="_blank"}, the catalogue of FOSS Android apps.

Wdyt?

Stadicus replied:
Good idea, me likes :+1:

VajraOfIndra commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+
+* Open Zeus and click on 'Get started'
+
+* Select 'Connect to an lnd node' at the top, and then '+ Add a new node'
```
Fixed

VajraOfIndra commented on mobile-app.md:
```diff
@@ -0,0 +1,159 @@
+
+* Select 'Connect to an lnd node' at the top, and then '+ Add a new node'
+
+* Click on 'Use Tor'
```
Fixed

VajraOfIndra commented:
> Just a few little wording changes, the overall guide is good to go! rocket

I implemented your changes. PTAL.

Couple of nits:
* Should "Get started" be in all caps like the other buttons?
* Nickname has a capital N.. is that a typo or on purpose?
* I used the code formatting for the buttons.. is that ok or do we prefer to simply use double-quotes (or single-quotes?) around the button name? in other words.. which one should we use in the example below?
  * `CONNECT A NODE`
  * 'CONNECT A NODE'
  * "CONNECT A NODE"

Stadicus commented:
> Should "Get started" be in all caps like the other buttons?

Good point, it's written in all caps

> Nickname has a capital N.. is that a typo or on purpose?

I wrote it with a capital N as it's written like this in the app. Depends on whether we want to reference the input field, or just give instructions to enter a nickname. Both are fine I think.

> I used the code formatting for the buttons.. is that ok or do we prefer to simply use double-quotes (or single-quotes?) around the button name? in other words.. which one should we use in the example below?

Up until now, I reserved code formatting for things you enter, or physical keyboard keys to press. But I guess that also works for touchscreen buttons. I'm just not a fan of single quotes, everything else is fine.

Stadicus commented:
LGTM! :100:

VajraOfIndra commented:
> > Should "Get started" be in all caps like the other buttons?
> 
> Good point, it's written in all caps

Ok, fixed
 
> > Nickname has a capital N.. is that a typo or on purpose?
> 
> I wrote it with a capital N as it's written like this in the app. Depends on whether we want to reference the input field, or just give instructions to enter a nickname. Both are fine I think.

Ok, I kept the capital N

> Up until now, I reserved code formatting for things you enter, or physical keyboard keys to press. But I guess that also works for touchscreen buttons. I'm just not a fan of single quotes, everything else is fine.

Ok, I changed to double-quotes. If one day we want to use code formatting for buttons (for readability), we could implement the change gloablly to all the relevant guides.

Will merge now! :tada:

uskrusader commented:
I'm stuck in the Zeus guide.  The QR code is massive and I can't shrink it.  Ctrl + - is not working in the putty termininal.  It works for most of my other windows, but not putty where that ginormous QR code is.  Cant fit it on screen 8-(
